buildSrc/src/main/groovy/net/minecraftforge/forge/tasks/checks/CheckATs.groovy

```diff
@@ -0,0 +1,304 @@
+                            key = entry.cls + ' ' + mtd.replace(' ', '')
+                            if (accessLevel(value.access) < accessStr(entry.modifier)) {
+                                if (lines.containsKey(key)) {
+                                    toRemove.add(key)
+                                } else if (!entry.existing.contains(key)) {
+                                    println('Added: ' + key)
+                                }
+                                entry.children.add(key)
+                            } else if (lines.containsKey(key)) {
+                                toRemove.add(key)
+                                println('Removed: ' + key)
+                            }
+                        }
+                        entry.existing.findAll { it !in entry.children }.each { println('Removed: ' + it) }
+                    }
+                } else if ('<init>' == entry.desc) { //Make all public non-abstract subclasses
+                    inheritance.each { tcls, value ->
+                        if (!value.methods || ((value.access & Opcodes.ACC_ABSTRACT) != 0)) return
+                        final parents = [] as Set
+                        String parent = tcls
+                        while (parent !== null && inheritance.containsKey(parent)) {
+                            parents.add(parent)
+                            def p = inheritance[parent]
+                            parent = p === null ? null : p.superName
+                        }
+                        if (parents.contains(entry.cls.replaceAll('\\.', '/'))) {
+                            value.methods.each { mtd, v ->
+                                if (mtd.startsWith('<init>')) {
+                                    def child = tcls.replaceAll('/', '\\.') + ' ' + mtd.replace(' ', '')
+                                    if (accessLevel(v.access) < 3) {
+                                        if (lines.containsKey(child)) {
+                                            toRemove.add(child)
+                                        } else if (child !in entry.existing) {
+                                            println('Added: ' + child)
+                                        }
+                                        entry.children.add(child)
+                                    } else if (lines.containsKey(child)) {
+                                        toRemove.add(child)
+                                        println('Removed: ' + child)
+                                    }
+                                }
+                            }
+                        }
+                    }
+                    entry.existing.findAll { it !in entry.children }.each{ println('Removed: ' + it) }
+                }
+            }
+
+            // Process normal lines, remove invalid and remove narrowing
+            else {
+                def jcls = inheritance.get(entry.cls.replaceAll('\\.', '/'))
```

Comment:
I've seen `entry.cls.replaceAll('\\.', '/')` enough times that I think it warrants being calculated once and put into a local variable. Perhaps, `.replaceAll('\\.', '/')` should also become a named function on its own (like, say, `toBinaryName`)?